### PR TITLE
New package: anonymous-communication

### PR DESCRIPTION
**Pull request by chrmod**
Todo:
* [x] fix linter
* [x] run tests in CI

### CODE REVIEW

**Comment by chrmod** [comment]
@philipp-classen I've made the test to pass locally. Please start the review while I will clean up the PR.

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+nodejs 16.14.1
```
Comment (smalluban) [inline .tool-versions:1]: Where do we need it? You can set a required node version in the `package.json`. 

Why not a 17.x? LTS? If so, we are not using node as a server, so we can use the newest version.

File: packages/anonymous-communication/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "mocha": "^9.2.2",
+    "rollup": "^2.70.1",
+    "rollup-plugin-polyfill-node": "^0.8.0",
+    "rollup-plugin-sourcemaps": "^0.6.3",
+    "sinon": "^13.0.1"
+  },
+  "dependencies": {
+    "pako": "^2.0.4"
+  },
+  "main": "index.js",
```
Comment (smalluban) [inline packages/anonymous-communication/package.json:30]: The preffered way is to set also `exports` field, but as we only distribute ES modules, we can skip the `main` field at all:

Node docs:

> set the main entry point for a package, it is advisable to define both ["exports"](https://nodejs.org/api/packages.html#exports) and ["main"](https://nodejs.org/api/packages.html#main) in the package’s [package.json](https://nodejs.org/api/packages.html#nodejs-packagejson-field-definitions) file

More here: https://nodejs.org/api/packages.html#package-entry-points

**Review by smalluban — CHANGES_REQUESTED** [review]

File: .tool-versions (hunk 1 above)
Comment (chrmod) [reply .tool-versions:1]: LTS is a moving target, so may break in future. 

I've added engines list in the package.json

File: packages/anonymous-communication/package.json (hunk 2 above)
Comment (chrmod) [reply packages/anonymous-communication/package.json:30]: :+1:

File: .tool-versions (hunk 1 above)
Comment (chrmod) [reply .tool-versions:1]: `.tool-versions` is used by `asdf` - a convenient tool to manage toolchain versions. we use it in many other projects, it works similar to `nodenv` or `nvm` but handle other toolchains

Comment (smalluban) [reply .tool-versions:1]: The `engines` are added in the package nested config, but this file is in the root...

Comment (smalluban) [reply .tool-versions:1]: I checked `asdf`, now I understand. However, I don't get this part:

> LTS is a moving target, so may break in future.

It seems to answer why we don't use LTS, but 16.x is LTS indeed :P

File: packages/anonymous-communication/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "karma-mocha-reporter": "^2.2.5",
+    "karma-rollup-preprocessor": "^7.0.8",
+    "karma-sinon": "^1.0.5",
+    "mocha": "^9.2.2",
+    "rollup": "^2.70.1",
+    "rollup-plugin-polyfill-node": "^0.8.0",
+    "rollup-plugin-sourcemaps": "^0.6.3",
+    "sinon": "^13.0.1"
+  },
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (chrmod) [reply packages/anonymous-communication/package.json:27]: I keep it here  in case of local testing - so you can run `npm ci` in the package folder.

**Review by smalluban — APPROVED** [review]

**Review by philipp-classen — APPROVED** [review]

**Pull request merged**